MHCollapsibleViewManager.m

```diff
@@ -0,0 +1,228 @@
+//
+//  MHCollapsibleViewManager.m
+//  This is a manager of a uiview and view controller
+//
+//  Created by Lizzy Randall on 5/6/15.
+//  Copyright (c) 2015 Lizzy Randall. All rights reserved.
+//
+
+#import "MHCollapsibleViewManager.h"
+#import "MHFilterLabel.h"
+
+@implementation MHCollapsibleViewManager
+
+//@synthesize tableView;
+//@synthesize viewController;
+@synthesize rowAnimation;
+
+@synthesize data;
+@synthesize cellStyles;
+@synthesize headerTitle;
+
+@synthesize expanded;
+@synthesize numOfRows;
+
+//Initialize with animation, title and tableView
+- (id)initalizeManager:(UITableViewRowAnimation)animation title:(NSString*) title tableView:(UITableView*)tableView
+{
+    self.headerTitle = title;
+    self.rowAnimation = animation;
+    //initialized first and then can be overwritten by controller
+    //tableView is needed to create cells to store types
+    [self setCellProperties:nil header:nil clickedCell:nil normalCell:nil tableView:tableView];
+    return self;
+}
+
+//called after Initializing Manager
+- (void) initializeData:(NSMutableArray*)dataFromController
```

Comment:
instead of creating an initializeData method like this we would normally override the automatically generated setter for the data property. You would also move the initialization of data to init as mentioned above.
eg
- (void)setData:(NSMutableArray *) {
[self willChangeValueForKey:@"data"];
[dataFromController enumerateObjectsUsingBlock:^(NSString *labelName, NSUInteger index, BOOL *stop) {
  [_data addObject:[MHFilterLabel constructLabel:labelName checked:NO]];
}];
[self didChangeValueForKey:@"data"];
self.numOfRows = _data.count;
}

Note: when overriding a setter you need to call the willChangeValueForKey: and didChangeValueForKey: methods incase someone needs to listen for changes in the property. Also as mentioned above you need to reference the ivar directly via _data =  instead of using self.data = (this is because when setting a value self.data = blah gets turned into [self setData:blah]; by the compiler which would be recursive. In this case you haven't done that you have just use [self.data addObject:] which the compiler turns into [[self data] addObject:]; which is not recursive